charts/datadog/templates/otel-configmap.yaml

```
@@ -0,0 +1,66 @@
+{{- if eq (include "should-enable-otel-agent" .root) "true" }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "agents-install-otel-configmap-name" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{ include "datadog.labels" . | nindent 4 }}
+  annotations:
+    checksum/confd-config: {{ tpl (toYaml .Values.datadog.confd) . | sha256sum }}
+    checksum/autoconf-config: {{ tpl (toYaml .Values.datadog.autoconf) . | sha256sum }}
+data:
+  otel-config.yaml: {{- if .Values.datadog.otelCollector.config }} {{ toYaml .Values.datadog.otelCollector.config | indent 4 }}
+  {{- else }}
+
+receivers:
```

I'm not sure that it works

usually for multiline in yaml you need to add `|` and keep some indentations 
```suggestion
  otel-config.yaml: {{- if .Values.datadog.otelCollector.config }} {{ toYaml .Values.datadog.otelCollector.config | indent 4 }}
  {{- else }} |

    receivers:
    ...
```